### DISCUSSION TITLE
How to open Link in new tab?

### DISCUSSION THREAD

**Discussion by naseef0** [vercel/next.js#15486] [Help] [answered]
![Screenshot 2020-07-26 at 10 17 54 AM](https://user-images.githubusercontent.com/45784156/88471655-47986580-cf29-11ea-887e-f35c2a4f2c72.png)

I need to open that link in new tab. how to do it?

i have tried this. 

> ` <Link
                href={{
                  pathname: "/product-details",
                  query: { id: product.id },
                }}
              >
                <a target="_blank" rel="noreferrer">
                  <img
                    className="img-fluid"
                    src={getImageUrlBySize(imageUrl, 3, mimeType)}
                    alt={productName}
                    onError={(e) => onImageError(e)}
                  />
                </a>
              </Link>`

But its not working.

**Comment by paambaati** [comment, marked as answer]
**EDIT**: See the [response from @Tylerian below](https://github.com/vercel/next.js/discussions/15486#discussioncomment-42331).

~~Adding the `passHref` prop to your `<Link>` component should solve this.~~

```
<Link href={{ pathname: "/product-details", query: { id: product.id }, }} passHref> <a target="_blank" rel="noreferrer"> <img className="img-fluid" src={getImageUrlBySize(imageUrl, 3, mimeType)} alt={productName} onError={(e) => onImageError(e)} /> </a> </Link>
```

~~Further reading — https://nextjs.org/docs/api-reference/next/link#forcing-link-to-expose-href-to-its-child~~

**Reply by Tylerian** [reply]
I wouldn't use Link component in this case. Opening in a new tab hinders any benefits Link component could provide, as it forces a full load of the site.

**Reply by paambaati** [reply]
@Tylerian You're right.

**Reply by DWMiller** [reply]
Please don't cross out your answer because it's often not advised. The question is what it is and you answered it. 

I recently had to do this, conditionally (I also don't think we should, but it's not my call), and your answer is the correct answer.

**Reply by srflp** [reply]
I would use Link component in this case, as it adds the current locale to the URL.

It's useful when you use i18n from next.js.

**Comment by Tylerian** [comment]
Simply put your navigation code inside an anchor with _blank as target (i.e: <a herf={\`/product-details/${product.id}\`} target="_blank">).

Using Link component here provides no performance benefit to the end user since you're requesting a fresh load of your site on a new browser tab.

**Reply by naseef0** [reply]
Hey @Tylerian  @paambaati ,
Thanks for your response. but actually... i need to see open link in new tab option. when i right click on an image which is wrapped by `Link` shown  above.

now its showing like this:

![Screenshot 2020-07-26 at 7 38 54 PM](https://user-images.githubusercontent.com/45784156/88481194-a684cb80-cf77-11ea-8aad-617f83c57dd3.png)

actually i need like this:

![Screenshot 2020-07-26 at 7 39 49 PM](https://user-images.githubusercontent.com/45784156/88481214-c7e5b780-cf77-11ea-9176-18505ab8dc4c.png)

**Reply by Tylerian** [reply]
So to make it clear, you want your image to open in a new browser tab when you right-click it, but to open on the same tab when you left-click it?

To do the former you simply have to wrap your image within an anchor. [Example](https://repl.it/repls/CumbersomeKnobbyYottabyte#pages/index.js) and [live code preview](https://cumbersomeknobbyyottabyte--five-nine.repl.co/).

```js
import Link from 'next/link';
 
function Page() {
  return (
    <div>
      <Link href="/index">
        <a>
          <img alt="Random image" src="https://source.unsplash.com/random/300x300" width="300" height="300" />
        </a>
      </Link>
    </div>
  )
}
 
export default Page
```

IMHO this question isn't even related with Next.js but simple HTML though.

**Reply by naseef0** [reply]
Hey, @Tylerian  

Thanks, Its working

**Reply by Tylerian** [reply]
Check the example above, it's all you need.

![Captura de pantalla 2020-07-26 a las 23 35 41](https://user-images.githubusercontent.com/1430805/88490055-fc1da000-cf98-11ea-827e-c84ee9fb27ce.png)

**Reply by naseef0** [reply]
@Tylerian  thank you brother.

**Reply by dangthien2704** [reply]
> So to make it clear, you want your image to open in a new browser tab when you right-click it, but to open on the same tab when you left-click it?
> 
> To do the former you simply have to wrap your image within an anchor. [Example](https://repl.it/repls/CumbersomeKnobbyYottabyte#pages/index.js) and [live code preview](https://cumbersomeknobbyyottabyte--five-nine.repl.co/).
> 
> ```js
> import Link from 'next/link';
>  
> function Page() {
>   return (
>     <div>
>       <Link href="/index">
>         <a>
>           <img alt="Random image" src="https://source.unsplash.com/random/300x300" width="300" height="300" />
>         </a>
>       </Link>
>     </div>
>   )
> }
>  
> export default Page
> ```
> 
> IMHO this question isn't even related with Next.js but simple HTML though.

Hey @Tylerian, Can I nested some div tags instead of just one image tag inside the a tag as below? Will it cause any problem?
![image](https://user-images.githubusercontent.com/67835018/123082923-c05db580-d449-11eb-9caf-cb1876c653d9.png)

**Reply by lingdocs** [reply]
The problem with this for me is that it means I have to deal with `<a>` tag styling, when I would really like to use just the `<Link>` component when I want something to look plain and avoid the link un-styling hassle. I've created a [feature request](https://github.com/vercel/next.js/issues/33561) for this.

**Reply by hypnoboutique** [reply]
> So to make it clear, you want your image to open in a new browser tab when you right-click it, but to open on the same tab when you left-click it?
> 
> To do the former you simply have to wrap your image within an anchor. [Example](https://repl.it/repls/CumbersomeKnobbyYottabyte#pages/index.js) and [live code preview](https://cumbersomeknobbyyottabyte--five-nine.repl.co/).
> 
> ```js
> import Link from 'next/link';
>  
> function Page() {
>   return (
>     <div>
>       <Link href="/index">
>         <a>
>           <img alt="Random image" src="https://source.unsplash.com/random/300x300" width="300" height="300" />
>         </a>
>       </Link>
>     </div>
>   )
> }
>  
> export default Page
> ```
> 
> IMHO this question isn't even related with Next.js but simple HTML though.

Arguably this _is_ next/react related. How does the `a`nchor magically get populated with the `Link` href? 🧐

**Reply by anguskeatinge** [reply]
@hypnoboutique 

The code example is slightly wrong here. 

The `Link` tag needs the prop `passHref` to pass the href to its child, nothing magic.

**Reply by hypnoboutique** [reply]
Thanks @anguskeatinge. I saw the prop in the type but noted that passing of passHref (`true`/`false`/`undefined`/no prop) doesn't seem to make any difference. The anchor becomes populated by the `Link` regardless. 🤷

**Reply by anguskeatinge** [reply]
My bad!!!

Just read through the implementation, turns out they treat `a` tags specially. Good spotting @hypnoboutique 🤘

```jsx

  // If child is an <a> tag and doesn't have a href attribute, or if the 'passHref' property is
  // defined, we specify the current 'href', so that repetition is not needed by the user
  if (props.passHref || (child.type === 'a' && !('href' in child.props))) {
    ...
    childProps.href = blah...
    ...
}
```
p.s. the canary `link.tsx` is confusing as hell because it is a mish mash of the new and old `Link` api, so above code is from `v11`

**Comment by deyvisonborges** [comment]
Create a **Anchor** component.

```ts
// Anchor.styles.ts
import styled from 'styled-components'

export const Anchor = styled.a`
  color: inherit;
  height: fit-content;
  text-decoration: none;
  width: fit-content;

  :visited,
  :hover,
  :active {
    color: inherit;
  }

  > * {
    color: inherit;
  }
`

```


```tsx
// Anchor.tsx
import Link from 'next/link'
import { ReactNode } from 'react'

type AnchorProps = {
  href: string
  children: ReactNode
  passHref?: boolean
  target?: '_blank' | '_self' | '_parent' | '_top'
}

export function Anchor({ href, passHref, target, children }: AnchorProps) {
  return (
    <Link href={href} passHref={passHref}>
      <S.Anchor href={href} target={target}>
        {children}
      </S.Anchor>
    </Link>
  )
}
```


Using

```tsx
  const render = () => (
  <Anchor href="https://example.com/new-page" target="_blank" >
       open in new page
  </Anchor>
)
```

**Comment by anguskeatinge** [comment]
If you want:

1. single click to follow link, but not refresh page
2. cmd+click to  open in a new tab

You need to follow @Tylerian 's comment: https://github.com/vercel/next.js/discussions/15486#discussioncomment-42380 and wrap an a tag with a Link tag.

```jsx
<div>
  <Link href='/blah' passHref>
    <a>
      <div>
        Component to link
      </div>
    </a>
  </Link>
</div>
```

This seems like a pretty common use case, e.g. _all_ of Vercel's links can be cmd+clicked, and almost every `Link` in my app wraps an `a` tag.

Would it be much effort to add a prop to the Link component, e.g. `cmdClickable` that handled this for us?

**Comment by nooobcoder** [comment]
Hi @paambaati @naseef0 @Tylerian @leerob from the release of Next v13, usage of anchor tag `<a></a>` inside `<Link/>` is disregarded, and flags errors.
Here is the [link from nextjs docs officially.](https://nextjs.org/docs/messages/invalid-new-link-with-extra-anchor)

Can anybody still tell me, how we can achieve the behavior of opening **Link** in new tab with the `<Link/>` component in NextJS.

Thanks.

**Reply by jeremybarthoux** [reply]
Hi @nooobcoder,
The `Link` extend `AnchorHTMLAttributes`, so you can pass prop like you did it on the `a` tag.
e.g. `<Link href={href} target="_blank" />`

**Reply by leerob** [reply]
If the link should open in a new tab, don't use `Link` and instead just a normal `a` tag 👍

**Reply by srflp** [reply]
@leerob unless you make use of automatically prepended locale identifiers ;) then it is easier and better to use `Link` with `target="_blank"`.

I would even say that this should be a recommended practice because it's future-proof (and i18n compatible)
